src/Features/Core/Portable/CodeFixesAndRefactorings/FixAllCodeAction.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Roslyn.Utilities;
+using FixAllScope = Microsoft.CodeAnalysis.CodeFixes.FixAllScope;
+
+namespace Microsoft.CodeAnalysis.CodeFixesAndRefactorings
+{
+    internal abstract class FixAllCodeAction : CodeAction
```

Comment:
All code in this file is existing code, moved up from sub-types.

Reply:
can you plz rename to AbstractFixAllCodeAction.  Add small amount of docs as well.   it might be potentially good to have this type lay out the inheritance hierarchy below it explaining what each subclass is responsible for.